### PR TITLE
tests: fix snap confine from core test to check the restart was done

### PR DESCRIPTION
**Pull request by sergiocazzolato**
This change is to check the restart info a the debug log instaed of
checking that in the journalctl that sometimes does not log any data.

The prevoius check is not enough to validate the restart has been done
because journalctl used to show log information which already contains
the string "DEBUG: restarting into".

Error:
https://travis-ci.org/snapcore/snapd/builds/235957914

### CODE REVIEW

**Comment by mvo5** [comment]
Thanks for this branch. It looks like the description does no longer match the actual diff - e.g. ` By counting the occurrencies we are sure a restart has been done.` is not something I see in the diff. Could you please update the description?

**Comment by fgimenez** [comment]
I think it would be a good idea to track why `SNAP_REEXEC` is being set to the wrong value in the first place (AFAIK this is in progress?), forcing it to be unset for this test would mask possible problems down the road IMO

File: tests/main/snap-confine-from-core/task.yaml [hunk 1]
```diff
@@ -19,8 +19,7 @@ execute: |
     fi
 
     echo "Ensure we re-exec by default"
-    snap list
-    journalctl | MATCH "DEBUG: restarting into"
+    /usr/bin/env SNAPD_DEBUG=1 snap list 2>&1 | MATCH "DEBUG: restarting into"
```
Comment (zyga) [inline tests/main/snap-confine-from-core/task.yaml]: Can you please drop the env and the list command (you can just say `SNAPD_DEBUG=1 snap`) and add a comment explaining what is going on there.

**Comment by chipaca** [comment]
closing as this has seen no activity for a month. Feel free to re-open when you're going to work on it again :-)

**Pull request closed without merging**